fix(http): wait for all XHR requests to finish before stabilizing application

alan-agius4 opened the pull request:
Previously, since the `HttpXhrBackend` is a singleton, the macrotask was created and completed only for the initial request since it was stored as in property in the class instance. This commit replaces this logic to create a macro task for every XHR request.
    
Closes #49730

alan-agius4 commented:
For visibility I did try to use `InitialRenderPendingTasks` https://github.com/angular/angular/commit/16555a4b43b96fa3560f918df6894dd4d9218ead but this caused `fakeAsync` tests to fail due to the pending timer. The timer was required due as the `InitialRenderPendingTasks` is promised based and completes when there are no pending tasks and there is no way to go back to a non complete state.

AndrewKushnir commented:
[Presubmit](https://fusion2.corp.google.com/presubmit/tap/523331313/OCL:523331313:BASE:523403044:1681225413337:b65c9f6c/targets).

AndrewKushnir reviewed (approved):
Looks great, thanks for the fix Alan 👍

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

jessicajaniuk reviewed (approved):
reviewed-for: public-api

alan-agius4 committed:
fix(http): wait for all XHR requests to finish before stabilizing application

Previously, since the `HttpXhrBackend` is a singleton, the macrotask was created and completed only for the initial request since it was stored as in property in the class instance. This commit replaces this logic to create a macro task for every XHR request.

Closes #49730

AndrewKushnir commented:
This PR was merged into the repository by commit 079f4bc1efc3bad10ac61d3819a923d1e971284d.
